# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceChild.cpp
# Commit: 89e396b2896e
# Full Hash: 89e396b2896ed7fc168a5b4ec2f8ead05f5c6802
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-03 04:20:58
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r 33ad5fa05209 -r 89e396b2896e netwerk/cookie/CookieServiceChild.cpp
--- a/netwerk/cookie/CookieServiceChild.cpp	Tue Jun 02 09:14:09 2020 +0000
+++ b/netwerk/cookie/CookieServiceChild.cpp	Tue Jun 02 09:15:24 2020 +0000
@@ -331,6 +331,9 @@
 
   nsCOMPtr<nsIPrincipal> principal = aDocument->EffectiveStoragePrincipal();
 
+  nsICookie::schemeType schemeType =
+      CookieCommons::PrincipalToSchemeType(principal);
+
   nsAutoCString baseDomain;
   nsresult rv = CookieCommons::GetBaseDomain(principal, baseDomain);
   if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -388,6 +391,10 @@
       continue;
     }
 
+    if (!CookieCommons::MaybeCompareScheme(cookie, schemeType)) {
+      continue;
+    }
+
     // if the nsIURI path doesn't match the cookie path, don't send it back
     if (!CookieCommons::PathMatches(cookie, pathFromURI)) {
       continue;